src/components/views/SupplierLookup.tsx

```diff
@@ -0,0 +1,198 @@
+import '../../styles/supplierLookup.css';
+import { Textfield } from '../base/Textfield';
+import '../../styles/globalbtn.css';
+import { useState, useEffect } from 'react';
+import { Supplier } from '../data/data';
+import { Link } from 'react-router-dom';
+import { searchSuppliers } from '../../utils/indexedDB';
+import SupplierTable from '../inputs/SupplierTable';
+
+interface SupplierLookupProps {
+  onClose: () => void;
+  onSupplierSelect: (supplier: Supplier) => void;
+}
+
+const SupplierLookup: React.FC<SupplierLookupProps> = ({
+  onClose,
+  onSupplierSelect,
+}) => {
+  const hardcodedSuppliers: Supplier[] = [
```

Comment:
You need to cleanup this component:
1. decide which state you keep, there is no reason to keep both supplier and filtered supplier, when you show only one in table.
2. decide how do you get suppliers. If you load them from store, there is no need to have hardcodedSuppliers in component.
3. please cleanup all stuff we already talked about: naming, no constants inside of component, no unused code (weird div in Header.tsx), no inline functions.
4. You have a bug with this code. If you navigate to certificate/0 on closing dialog, edit won't work properly. Please just close dialog without navigating. Hint: you can use same handler as on select, with passing null value and then in handler do check if supplier was selected or not.

Reply:
I also wrote comments on this. Switch to store, search supliers from store (indexDB actually), get rid of hardcoded suppliers, and have only one state holding result from database.